explorer/explorer.go

```diff
@@ -16,6 +22,50 @@ var (
 	ErrNotAvailable = errors.New("current OS not supported")
 )
 
+type result struct {
+	file  interface{}
+	error error
+}
+
+type Explorer struct {
+	id    int32
+	mutex sync.Mutex
+
+	// explorer holds OS-Specific content, it varies for each OS.
+	*explorer
+}
+
+// active holds all explorer currently active, that may necessary for callback functions.
+//
+// Some OSes (Android, iOS, macOS) may call Golang exported functions as callback, but we need
+// someway to link that callback with the respective explorer, in order to give them a response.
+//
+// In that case, a construction like `callback(..., id int32)` is used. Then, it's possible to get the explorer
+// by lookup the active using the callback id.
+//
+// To avoid hold dead/unnecessary explorer, the active will be removed using `runtime.SetFinalizer` on the related
+// Explorer.
+var active = sync.Map{} // map[int32]*explorer
+
+// NewExplorer creates a new Explorer for the given *app.Window.
+// The given app.Window must be unique and you should call NewExplorer
+// once per new app.Window.
```

Comment:
Can we do something when the user calls `NewExplorer` more than once with the same window? I suppose we'd need to iterate the active map searching for the window, which is inherently somewhat racy. Just curious what your thoughts are on ways to prevent the user from making this mistake.

Reply:
It will require another `map.Sync` (or `map`) since the current one doesn't hold the `Explorer` (just the internal `explorer`). But, if it keeps the `Explorer` it'll never be released by GC.

Reply:
Yes, I agree that a way to define custom ops in Gio would be really nice, and would greatly simplify this. :D We'll have to wait to hear Elias' view on that though. I guess for now we can leave the API as-is. It's tricky to know what the "right" way to handle this kind of misuse would be.